packages/next/next-server/server/next-server.ts

```diff
@@ -339,6 +362,71 @@ export default class Server {
       },
     ]
 
+    const { redirects = [], rewrites = [] } = this.customRoutes
+
+    const getCustomRoute = (
+      r: { source: string; destination: string; statusCode?: number },
+      type: 'redirect' | 'rewrite'
+    ) => ({
+      ...r,
+      type,
+      matcher: customRoute(r.source),
+    })
+
+    const customRoutes = [
+      ...redirects.map(r => getCustomRoute(r, 'redirect')),
+      ...rewrites.map(r => getCustomRoute(r, 'rewrite')),
+    ]
+
+    routes.push(
+      ...customRoutes.map((r, idx) => {
+        return {
+          match: r.matcher,
+          fn: async (req, res, params, parsedUrl) => {
+            let destinationCompiler = pathToRegexp.compile(r.destination)
+            let newUrl = destinationCompiler(params) // /blog/123
+            let newParams = params // { id: 123 }
+            let statusCode = r.statusCode
+            const followingRoutes = [...customRoutes].splice(
+              idx,
+              customRoutes.length - idx
+            )
+
+            for (const followingRoute of followingRoutes) {
+              if (r.type === 'redirect' && followingRoute.type !== 'redirect') {
+                continue
+              }
+
+              if (isDynamicRoute(followingRoute.destination)) {
+                throw new Error(
+                  `Routing a custom route to a dynamic page is currently not supported. This behavior is being investigated`
+                )
+              }
+
+              const curParams = followingRoute.matcher(newUrl)
+
+              if (curParams) {
+                destinationCompiler = pathToRegexp.compile(
+                  followingRoute.destination
+                )
+                newUrl = destinationCompiler(newParams)
+                statusCode = followingRoute.statusCode || 307
```

Comment:
```suggestion
                statusCode = followingRoute.statusCode
```

This is unnecessary, right?